src/seedu/addressbook/Main.java

```diff
@@ -107,6 +108,7 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            storage.checkStorageFileExists();
```

Comment:
In this context, `checkStorageFileExists()` seems to be on a lower abstraction level compared to the rest of the methods.
Since `checkStorageFileExists()` is the responsibility of `StorageFile` it would be ideal for `checkStorageFileExists()` to be called in `StorageFile` methods as per SLAP, in this case inside `storage.save(...)`.